cognite/client/config.py

```diff
@@ -82,13 +82,16 @@ def apply_settings(self, settings: dict[str, Any] | str) -> None:
 
         loaded = load_resource_to_dict(settings)
 
+        if not set(loaded).issubset(set(self.__dict__)):
+            raise ValueError(
+                f"Invalid keys provided for global_config, no settings applied: {set(loaded) - set(self.__dict__)}"
+            )
+
         for key, value in loaded.items():
-            if not hasattr(global_config, key):
-                raise ValueError(f"Invalid key in global config: {key}")
             if key == "default_client_config":
-                if isinstance(value, ClientConfig):
+                if isinstance(value, ClientConfig) or value is None:
                     self.default_client_config = value
-                else:
+                elif value:
```

Comment:
If someone passes in a value in disagreement with the type annotation, which are handled in the `if`, I'd say it's better to let `ClientConfig.load` decide if it can be accepted or not:
```suggestion
                else:
```

Reply:
should be fine that it will raise `TypeError(f"Resource must be json or yaml str, or dict, not {type(resource)}")` so reverted to not doing additional checks